### DISCUSSION TITLE
Sending 204 when event/stream is done?

### DISCUSSION THREAD

**Discussion by pamelafox** [pallets/flask#5130] [Q&A] [answered]
I'm using Flask to send server-sent events:

https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events-intro

I'm using a Response() with a generator function and event/stream mime-type:
https://github.com/pamelafox/azure-chatgpt-flask-container/blob/59194c08747c50ba26a517582579fc10970a0381/src/flaskapp/chat.py#L32

The issue is that I want to be able to tell the client that the stream is done (generator is exhausted), and the spec says that sending a 204 will signal that. I don't see an obvious way to send a 204 using this way of using Response, however. Is that possible?

(My workaround is to send a "bye-bye-bye" event and have the client listen for that, but it'd be cleaner for the client to not have to do the closing.)

Thanks!

**Comment by ThiefMaster** [comment]
usually you can just ignore it. if you don't plan to make things stateful by letting a client specify the last event they got (last-event-id header) and then resending anything they missed since they disconnected it won't make sense to implement any other reconnect-related logic either.

depends on your usecase of course...

**Comment by RonnyPfannschmidt** [comment, marked as answer]
As per the spec, the 204 is to be used after the client reconnects

The expectation is that the client keeps reconnecting, and the server can use a204 on the next connection

**Reply by davidism** [reply]
To expand on this, there's no way to send a new status code after sending the response body, that's HTTP (not Flask). 204 would be used to signal that there's no content when the request is started, so it would only be sent on subsequent requests if there's no new data.

**Comment by pamelafox** [comment]
Ahh, right, of course, that makes sense. Thanks all!
